### PR TITLE
CAMEL-15689 - Add camel-atlasmap component

### PR DESCRIPTION
**Pull request by igarashitm**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
  - https://issues.apache.org/jira/browse/CAMEL-15689
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
  - This is to migrate camel-atlasmap component from AtlasMap project repository into Apache Camel upstream. It has been maintained in [AtlasMap project](https://github.com/atlasmap/atlasmap) until it is stabilized. 
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by oscerd — APPROVED** [review]
LGTM. Lets wait for others reviewers. Thanks @igarashitm !

**Review by davsclaus — APPROVED** [review]

File: components/camel-atlasmap/src/main/java/org/apache/camel/component/atlasmap/AtlasMapEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+                    break;
+                default:
+                    LOG.info("{}: Document='{}(ID:{})', path='{}'",
+                            audit.getMessage(), audit.getDocName(), audit.getDocId(), audit.getPath());
+            }
+        }
+        if (!errors.isEmpty()) {
+            StringBuilder buf = new StringBuilder("Errors: ");
+            errors.stream().forEach(a -> buf.append(
+                    String.format("[%s: Document='{}(ID:{})', path='%s'], ",
```
Comment (aldettinger) [inline components/camel-atlasmap/src/main/java/org/apache/camel/component/atlasmap/AtlasMapEndpoint.java:220]: Do we have a mix here where last 2 params are ignored ?

**Review by aldettinger — APPROVED** [review]
Just a small question above but otherwise looks good to me. Well done @igarashitm :)

File: components/camel-atlasmap/src/main/java/org/apache/camel/component/atlasmap/AtlasMapEndpoint.java (hunk 1 above)
Comment (igarashitm) [reply components/camel-atlasmap/src/main/java/org/apache/camel/component/atlasmap/AtlasMapEndpoint.java]: Doh good catch, thanks! fixed.

**Pull request merged**